docs(vault) - update sections for vault superflags (#133)
dgraph-io/dgraph-docs: content/enterprise-features/access-control-lists.md

The ACL secret can be saved on [Hashicorp Vault](https://www.vaultproject.io/) server instead saving the secret on the local file system.

### Configuring a Hashicorp Vault Server

These prerequisites are required when setting up [Hashicorp Vault](https://www.vaultproject.io/) server for use with Dgraph:

To use Hashicorp Vault, these prerequisites must for the Vault Server.
1. Accessible from Dgraph Alpha and configured using URL `http://fqdn[ip]:port`.
2. Enable [AppRole Auth method](https://www.vaultproject.io/docs/auth/approle) and enable [KV Secrets Engine](https://www.vaultproject.io/docs/secrets/kv).
3. Save the 256-bits (32 ASCII characters) long ACL secret in a KV Secret path.  For example, you can upload this below to KV Secrets V2 path of `secret/data/dgraph/alpha`:
   ```json
   {
     "options": {
       "cas": 0
     },
     "data": {
       "hmac_secret_file": "12345678901234567890123456789012"
     }
   }
   ```   
4. A role with an attached policy that grants access to the secret.  For example, this policy would grant access to `secret/data/dgraph/alpha`:
   ```hcl
   path "secret/data/dgraph/*" {
     capabilities = [ "read", "update" ]
   }
   ```

> 5. The `role-id` and corresponding `secret-id` must must be generated and copied over to local files.

Author: This is backwords. Vault Server generates the role-id and secret-id.

> 5. The `role-id` and corresponding `secret-id` must must be generated and copied over to local files.

Author: This cannot be done because:
1. role-id and secret-id are generated on the Vault Server.
2. they are then copied to text files for Dgraph Alpha Nodes.
3. the path is symbolic path inside Vault Server KV Secret, and is used in http url path to reference the symbolic path that represents context inside Hashicorp vault (one-to-one http<->kv-path)

> 5. The `role-id` and corresponding `secret-id` must must be generated and copied over to local files.

Author: Given the style suggested, I made this below. I feel this has too much verbosity:

> On the Hashicorp Vault server, generate the `role_id` and corresponding `secret_id` whose policy can access the KV Secret path, and copy them over to local files that will be used by Dgraph Alpha nodes like `./dgraph/vault/role_id` and `./dgraph/vault/secret_id`.

      DGRAPH_ALPHA_SECURITY: whitelist="10.0.0.0/8,172.0.0.0/8,192.168.0.0/16"
    image: dgraph/dgraph:{{< version >}}
    ports:
      - 8080:8080
    volumes:
      - ./role_id:/dgraph/vault/role_id
      - ./secret_id:/dgraph/vault/secret_id
  zero1:
    command: dgraph zero --my=zero1:5080 --replicas 1 --raft idx=1
    container_name: zero1
    image: dgraph/dgraph:{{< version >}}
```

In this example, you will also need to configure a [Hashicorp Vault](https://www.vaultproject.io/) service named `vault` in the above `docker-compose.yaml`, and then run through this sequence:

1. Launch `vault` service: `docker-compose up --detach vault`
2. Unseal and Configure `vault` with the required prerequisites (see [Configuring a Hashicorp Vault Server](#configuring-a-hashicorp-vault-server)).
3. Save role-id and secret-id as `./role_id` and `secret_id`
4. Launch Dgraph Zero and Alpha: `docker-compose up --detach`

### Example using Kubernetes Helm Chart with Hashicorp Vault configuration

If you deploy Dgraph on [Kubernetes](https://kubernetes.io/), you can configure the ACL feature using the [Dgraph Helm Chart](https://artifacthub.io/packages/helm/dgraph/dgraph).

> The next step is that we need to create a [Helm](https://helm.sh/) chart config values file, e.g. `dgraph_values.yaml`.

Reviewer: ```suggestion
The next step is that we need to create a [Helm](https://helm.sh/) chart config values file, such as `dgraph_values.yaml`.
```

```yaml
## dgraph_values.yaml
alpha:
  configFile:
    config.yaml: |
      vault:
        addr: http://vault-headless.default.svc.cluster.local:9200
        acl_field: hmac_secret_file
        acl_format: raw
        path: secret/data/dgraph/alpha
        role_id_file: /dgraph/vault/role_id
        secret_id_file: /dgraph/vault/secret_id
      security:
        whitelist: 10.0.0.0/8,172.0.0.0/8,192.168.0.0/16‘
```

> The [Hashicorp Vault](https://www.vaultproject.io/) service will need to be installed and available.  This can be installed using the [Hashicorp Vault Helm Chart](https://www.vaultproject.io/docs/platform/k8s/helm) and can be configured to [auto unseal](https://learn.hashicorp.com/collections/vault/auto-unseal) so that the service is immediately available after deployment.

Reviewer: ```suggestion
To set up this chart, the [Hashicorp Vault](https://www.vaultproject.io/) service must be installed and available. You can use the [Hashicorp Vault Helm Chart](https://www.vaultproject.io/docs/platform/k8s/helm) and configure it to [auto unseal](https://learn.hashicorp.com/collections/vault/auto-unseal) so that the service is immediately available after deployment.
```
